### ISSUE TITLE
Unable to find locally built image

### ISSUE THREAD

**Issue by brianfoshee** [docker/compose#1568]
I have a locally built image (built with `docker build -t project/heroku-base ./project-heroku-base`) that version 1.3 of docker-compose is not finding during the build process for another image. 

I can build with `docker build app ./app`, and with version 1.2 of docker-compose I can build with `docker-compose build app`.

```
$ docker-compose build app
Building app...
Step 0 : FROM project/heroku-base:latest
Pulling repository project/heroku-base
Service 'app' failed to build: Error: image project/heroku-base:latest not found
```

```
$ docker images
REPOSITORY          TAG                 IMAGE ID            CREATED             VIRTUAL SIZE
project/heroku-base latest              ebe895ea759a        10 minutes ago      1.325 GB
mysql               latest              d63d4723d715        2 days ago          283.5 MB
debian              jessie              bf84c1d84a8f        2 days ago          125.2 MB
heroku/cedar        14                  89fc314a0a6e        4 weeks ago         1.242 GB
busybox             latest              8c2e06607696        8 weeks ago         2.433 MB
```

```
$ docker-compose -v
docker-compose version: 1.3.0
CPython version: 2.7.9
OpenSSL version: OpenSSL 1.0.1e 11 Feb 2013
$ docker version
Client version: 1.7.0
Client API version: 1.19
Go version (client): go1.4.2
Git commit (client): 0baf609
OS/Arch (client): linux/amd64
Server version: 1.7.0
Server API version: 1.19
Go version (server): go1.4.2
Git commit (server): 0baf609
OS/Arch (server): linux/amd64
```

**Comment by aanand** [comment]
Thanks for reporting. I can reproduce this with Compose 1.3.0 and Docker 1.7.0, but the following combinations do *not* exhibit the problem:

- Compose 1.3.0 and Docker 1.6.2
- Compose 1.2.0 and Docker 1.7.0

```
$ cat repo/Dockerfile
FROM busybox

$ cat app/Dockerfile
FROM user/repo:latest

$ cat docker-compose.yml
app:
  build: ./app
```

```
$ docker build -t user/repo ./repo
Sending build context to Docker daemon 2.048 kB
Sending build context to Docker daemon
Step 0 : FROM busybox
 ---> 8c2e06607696
Successfully built 8c2e06607696

$ docker-compose build app
Building app...
Step 0 : FROM user/repo:latest
Pulling repository user/repo
Service 'app' failed to build: Error: image user/repo:latest not found
```

When using Docker 1.7.0, here's the (working) request sent by the Docker 1.7.0 client:

```
POST /v1.19/build?cgroupparent=&cpuperiod=0&cpuquota=0&cpusetcpus=&cpusetmems=&cpushares=0&dockerfile=Dockerfile&memory=0&memswap=0&rm=1&t=app
```

Here's the (working) request sent by Compose 1.2.0:

```
POST /v1.15/build?pull=True&nocache=False&q=False&t=1568_app&forcerm=False&rm=True
```

Here's the (non-working) request sent by Compose 1.3.0:

```
POST /v1.18/build?pull=True&nocache=False&q=False&t=1568_app&forcerm=False&rm=True
```

Perhaps the problem is isolated to a particular API version or range of versions (1.15 and 1.19 work, 1.18 doesn't).

**Comment by aanand** [comment]
Right. After some digging, I've found the problem:

The stable version of docker-py has a bug in `build()` where `pull=True` by default (https://github.com/docker/docker-py/issues/622). This means that on every `docker-compose build`, we're asking the daemon to pull the image, even if it exists locally. This was fixed in https://github.com/docker/docker-py/pull/624.

The reason it works with Docker 1.6.2 is actually because of a bug in the Docker API where the `pull` parameter is ignored if it's not set to `"1"`: https://github.com/docker/docker/issues/13631. Docker 1.7.0 longer ignores the `pull` parameter. So this bug has appeared because another bug was fixed.

I've fixed it by explicitly setting `pull=False` in Compose: #1570
